src/main/java/org/wildfly/plugin/tools/client/ToolsModelControllerClientConfiguration.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.plugin.tools.client;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+import javax.net.ssl.SSLContext;
+import javax.security.auth.callback.CallbackHandler;
+
+import org.jboss.as.controller.client.ModelControllerClientConfiguration;
+import org.wildfly.plugin.tools.security.ClientCallbackHandler;
+import org.wildfly.security.SecurityFactory;
+
+/**
+ * A configuration used to connect a {@link org.jboss.as.controller.client.ModelControllerClient} or used to connect a
+ * CLI {@code CommandContext}
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public class ToolsModelControllerClientConfiguration implements ModelControllerClientConfiguration {
```

Comment:
Nothing to stop this from being merged, but we should consider adding a `DelegateModelControllerClientConfiguration` to WildFly Core. It allows for easier integration with different versions without needing to override additional methods.